### ISSUE TITLE
TSML v3.11 UI Issue with Addresses for Online Meetings

### ISSUE THREAD

**Issue by tech2serve** [code4recovery/12-step-meeting-list#443]
TSML v3.11 has a UI issue with online meetings and how addresses (either specific or approximate) are displayed on a single meeting page. 

These are the three general situations:

1. A hybrid meeting (for which the the display of specific address is necessary).
2. A natively online meeting (for which an approximate address is acceptable, but has limited value).
3. An online meeting in lieu of a physical meeting that is temporarily inactive (for which it is not desirable to show an address that could confuse a user).

The trick is that with situation 2, the approximate address should show normally, while in situation 3, the specific address should be lined through or otherwise obscured. Yet, we don’t have an easy way to differentiate those with CSS. 

We defined `attendance_option` as a single field and it doesn’t cleanly represent situation 3 above. TSML does have `tsml-type-*` css classes that could potentially be used to line through the specific address, but it is currently assigned to all meetings that have `attendance_option=‘online’` and doesn’t differentiate natively online meetings from those established to take the place of a temporarily inactivated in-person meetings. 

I see three options to address this:
1. Rework the code to ensure `tsml-type-tc` does not get assigned to the natively online meetings. This can then be used to line through the specific address described in situation 3 above.
2. Adopt the test included in the [fix-lineout-single-page-tc](https://github.com/code4recovery/12-step-meeting-list/tree/fix-lineout-single-page-tc) branch, or a version that generates new css class to address situation 3 above.
3. Remove the visible addresses on the `single-meeting.php` page for any meeting with `attendance_option=‘online’`. 

Option 3 hides non-critical information from the end user without forcing the admin to remove it from the data. The meeting list page would still include the approximate location for natively online meetings. This would be best as part of a broader UI redesign.

Option 1 would probably be best handled by @brianw-area24 who made most of the `attendance_option` code changes. The merit of this approach increases if the team decides Option 3 is not suitable. 

Option 2 is a quick fix that could be adopted in the interim. 

Thoughts of @code4recovery/tsml-contributors?

**Comment by OSAwebsite** [comment]
I don't think I am a contributor in the sense that I write code for TSML.
But my thought about this is: keep it simple and transparent. Don't try to solve data issues in code.
In other words: give the admins an address option that covers the 3 scenarios and have the code work with that.
How would that look? Leave what you have now in place. Add a single field with a dropdown/radio button choice that indicates what the admin of the listings wants to do with the address. The code only has to react to that and no complicated and possibly confusing branching is required.

Just my five cents. Good luck!

**Comment by Grateful-Viv** [comment]
If you could add an attendance_option code for natively online meetings, then two things could happen when looking at the meeting detail.

1. Remove the strikeout on the location
2. No longer state there are "xx" meetings at this location.

**Comment by joshreisner** [comment]
I believe this has been closed now with yesterday's release of `3.11.3`

**Comment by Grateful-Viv** [comment]
It's striking out the location for all online meetings. It doesn't make sense to do that for ones that are natively online.

I put in a fix with v3.11.2 by adding the following CSS which I removed with the update.

```
body.attendance-online p.location-address {
	text-decoration: line-through;
}
```

What other changes were made in this release?

**Comment by Grateful-Viv** [comment]
> I believe this has been closed now with yesterday's release of `3.11.3`

It’s working correctly. Thank you.
